Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -80,6 +79,8 @@ public function getWorkspaceMetadata(ContentRepositoryId $contentRepositoryId, W
 
     /**
      * Update/set title metadata for the specified workspace
+     *
+     * NOTE: The workspace privileges are not evaluated for this interaction, this should be done in the calling side if needed
```

Comment:
right.. I mean we could inject the AuthService here and evaluate, but it creates a weird dependency graph and since this is a rather low level method (and no high risk) I went for the easy way out :)

Reply:
low level as in: not required for the _regular_ interaction with a CR.
I agree that it at least feels wrong not to have the ACL enforced when changing the title/description metadata of a workspace but I don't think that it is an issue in reality and that the bidirectional dependency just to solve this case would be worse. But maybe others have a different opinion!?

Reply:
cc @kitsunet

Reply:
I am torn here between "oh my, we put so much on the shoulders of the end user with all this" and "yes, this is modular and flexible to use". I think I lean towards doing it like this and putting it on the implementer to decide if they want/need protection in the interaction. I just wonder because that makes for a weird collection of methods in the API, what about the ones that do _not_ explicitly state the are not protected, like e.g. getPersonalWorkspaceForUser

Maybe we should put this up as a general info/warning that nothing in this class is protected and you should do the respective checks before using it's APIs?

Reply:
> I just wonder because that makes for a weird collection of methods in the API

Yes, I agree.
I mean: AFAIS it's really "just" about these two methods, `setWorkspaceTitle` and `setWorkspaceDescription` as they are no longer a CR core concern (tbh: maybe we should have _kept_ it one..).

> what about the ones that do not explicitly state the are not protected

In the current implementation there is no restriction to _reading_ workspaces, everybody can read all information about all workspaces (as long as the UI allows them).

I think we can justify the current state with this argumentation: The CR core ensures the _important_ protections (which user can manage workspaces, read/edit nodes from certain workspaces) and it's the responsibility of the Application/UI layer to hide the rest – but it would not mean a security breach if they didn't..

> Maybe we should put this up as a general info/warning that nothing in this class is protected

Yes, good point. I'll add something like the above statement to the class

Reply:
Fixed via

- [TASK: Split mighty WorkspaceService into `WorkspaceMetadataAndRoleR…](https://github.com/neos/neos-development-collection/pull/5298/commits/952932700344fbe63c1a516404582c1a054433ea) 
[9529327](https://github.com/neos/neos-development-collection/pull/5298/commits/952932700344fbe63c1a516404582c1a054433ea)
- [FEATURE: Check permissions for operations inside WorkspaceService](https://github.com/neos/neos-development-collection/pull/5298/commits/4f07965b295b93d105214bac67187ec893b34a24)

the `WorkspaceService` got to bloated and now the `WorkspaceService` and the Cr Auth Service use a low level INTERNAL `WorkspaceMetadataAndRoleRepository` for this task. That allows us to require the Cr Auth Service _inside_ the `WorkspaceService` without any lazy object hacks.

The reason why bastian and me rediscussed this was because well `setWorkspaceTitle` is not thaaat important but having security for `assignWorkspaceRole` would be nice and it now is that way :)